### PR TITLE
[FIX] #157 Check that the use_parent_address field exists before using i...

### PR DESCRIPTION
**Pull request by ccomb**
...t. Also revert the hardcoded layout bloc to ease merge from upstream.

Fix for OCB issue #157

### CODE REVIEW

**Comment by petrus-v** [comment]
:+1:  (review only not tested)

File: openerp/addons/base/res/res_partner.py [hunk 1]
```diff
@@ -87,6 +70,9 @@ def fields_view_get_address(self, cr, uid, arch, context={}):
                     tree = etree.fromstring(v)
                     # needs to apply modifiers
                     for childnode in tree.getchildren():
+                        if 'use_parent_address' in self._all_columns:
```
Comment (pedrobaeza) [inline openerp/addons/base/res/res_partner.py]: But this is going to apply the modifier to each element found, including div and so on, doesn't it?

Comment (petrus-v) [inline openerp/addons/base/res/res_partner.py]: I don't think so we are here on children of `adress_format` css class:
``doc.xpath("//div[@class='address_format']")``

**Comment by pedrobaeza** [comment]
And what about putting the attrs in the div tag?

**Comment by ccomb** [comment]
Good idea, I'm trying

**Comment by ccomb** [comment]
Nope, that does'nt work on the div.

**Comment by pedrobaeza** [comment]
Thanks for the patch.

:+1:

**Comment by guewen** [comment]
Are we guaranteed that the field `use_parent_address` exists in the view?

**Comment by ccomb** [comment]
Good remark, my check is done on the model, not the view. I'm looking again.

**Comment by ccomb** [comment]
@guewen you were right, it should be better now.  Doing the same for v8

**Comment by guewen** [comment]
:+1:

**Comment by ccomb** [comment]
@pedrobaeza sorry, could you remerge and include the latest commit ? I had removed the check of the type filed in a previous commit.
This does not hurt, but having a `<br/>` tag with a "attrs" is not very useful...

I've just fixed on v8 as well before merge.

**Comment by pedrobaeza** [comment]
OK, I will amend the commit in a couple of hours.

**Comment by pedrobaeza** [comment]
I have made it in a new commit (https://github.com/OCA/OCB/commit/0ae923b7307ae0f2656fc64711f9e6e2d4d5a1ce) to avoid messing people that have made pulls in between.

**Comment by ccomb** [comment]
Thank you!

**Pull request merged**